data_structures/k_tree/k_tree.py

```diff
@@ -0,0 +1,79 @@
+class Node:
+    """defines each node used to build the k-ary tree"""
+    def __init__(self, val, input_list=[]):
+        self.val = val
+        self.children = []
+        for child in input_list:
+            self.children.append(Node(child))
+
+    def __repr__(self):
+        # return 'node children: {} '.format(lambda x: for x in self.children: return x.val)
+        return str(self.val) 
+
+    def __str__(self):
+        return 'node value: {}'.format(self.val)
+
+
+class Ktree:
+    """defines k-ary tree"""
+    def __init__(self, val, input_list=[]):
+        """defines root node"""
+        # if input_list is not list:
+        #     return KeyError
+        node = Node(val, input_list)
+        self.root = node
+
+    def __repr__(self):
+        """returns a string of type and address"""
+        # return 'node children: {} '.format(lambda x: for x in self.children: return x.val)
+        # return '<{} at {}>'.format(type(self, hex(id(self))))
```

Comment:
If you are overwriting this method, you need to return some string or it will break.